### PR TITLE
[FIX] l10n_eg_edi_eta: send invoices with individual partner

### PR DESCRIPTION
**Pull request by ThomasBeckers**
Before the fix it was not possible to create an invoice with an individual partner and send it to the government. Also the 'type' of the receiver is a required property to send to the government. There is no mention of it being optional in the API (https://sdk.invoicing.eta.gov.eg/documents/invoice-v1-0/#receiver).

Now it's possible to send such invoice and the 'type' of the receiver is always sent.

opw-2979515

### CODE REVIEW

File: addons/l10n_eg_edi_eta/models/account_edi_format.py [hunk 1]
```diff
@@ -185,7 +185,8 @@ def _l10n_eg_validate_info_address(self, partner_id, issuer=False, invoice=False
         fields = ["country_id",
                   "state_id", "city", "street",
                   "l10n_eg_building_no"]
-        if (invoice and invoice.amount_total >= invoice.company_id.l10n_eg_invoicing_threshold) or self._l10n_eg_get_partner_tax_type(partner_id, issuer) != 'P':
+        partner_type = self._l10n_eg_get_partner_tax_type(partner_id, issuer)
+        if partner_type != 'P' and invoice and invoice.amount_total >= invoice.company_id.l10n_eg_invoicing_threshold:
```
Comment (jco-odoo) [inline addons/l10n_eg_edi_eta/models/account_edi_format.py:189]: @ThomasBeckers Why are these changes necessary?  Because the first and should be or.   I don't think we need "invoice and" because invoice is always there.

Comment (ThomasBeckers) [reply addons/l10n_eg_edi_eta/models/account_edi_format.py:189]: I included this because it was in the commit BAA is working on. If you are not confident with it I can remove it. The most important for me part is the second change with "type".

Comment (bakboka) [reply addons/l10n_eg_edi_eta/models/account_edi_format.py]: @jco-odoo this change is because individuals do not need to have a vat number. So we want to enforce the vat only for businesses that have an invoice greater than the threshold
Regarding the "invoice and", I left it there since it was in the original code. I assumed it came from a bug fix

Comment (jco-odoo) [reply addons/l10n_eg_edi_eta/models/account_edi_format.py]: @bakboka I assume it is the method being called for cases where you do not have invoice.  The logic in the documentation (https://sdk.invoicing.eta.gov.eg/documents/invoice-v1-0/#receiver :  https://sdk.invoicing.eta.gov.eg/documents/invoice-v1-0/#receiver)is that you do not need the vat number if you are Person buyer and below the treshold, so you need one when you are not Person or exceed the treshold.  So, the documentation is wrong the way I read it?  @bakboka

Comment (bakboka) [reply addons/l10n_eg_edi_eta/models/account_edi_format.py]: @jco-odoo I think the documentation might be outdated but lets wait for @nadaseder to come back from vacation to merge that particular line

**Comment by jco-odoo** [comment]
@robodoo r+

**Comment by nadaseder** [comment]
@bakboka @jco-odoo 
This is to confirm that any invoice issued to an individual, and the total amount is less than the threshold (set on the company settings) should be sent to the authority without the need for a national ID number. 

Please note that if the user fills the national ID number and the invoice is below the threshold, Odoo should send the national ID number to the ETA portal. If it was not filled by the Odoo user, Invoice should also be processed normally and sent to the ETA. 

Any invoice issued to a company for any amount needs the TAX ID of the company. 
Any invoice issued to an individual for an amount that exceeds the threshold should mention the individual's national ID number. 

Hope this clears it

Thanks,
